### ISSUE TITLE
TvOS : Expo 51 a white screen appears

### ISSUE THREAD

**Issue by stdiodavid** [expo/expo#29600]
### Summary

1.npx create-expo-app etv51 -e with-router-tv
2.cd etv51
3.npm run ios

<img width="1259" alt="image" src="https://github.com/expo/expo/assets/144206414/9b15a578-da54-4d81-8f0b-6ffefa0d9219">

### What platform(s) does this occur on?

iOS

### SDK Version

51

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.20.3 - /usr/local/opt/node@18/bin/node
      Yarn: 1.22.22 - /usr/local/bin/yarn
      npm: 10.7.0 - /usr/local/opt/node@18/bin/npm
      Watchman: 2024.05.06.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.8 => 51.0.11
      expo-router: ~3.5.12 => 3.5.15
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native-tvos:  0.74.1-0
      react-native-web: ~0.19.10 => 0.19.12
    Expo Workflow: managed

### Minimal reproducible example

no need

**Comment by douglowder** [comment]
This is the same issue as #29607 .

In this case, the asset registry issue is causing the font in `app/_layout.tsx` to not load correctly, which means that the splash screen never hides.

The correct fix is to fix the asset-registry dependency.

Quick workaround is to detect the font error and allow the app to continue to work, but also show the font loading error as a warning:

```patch
diff --git a/app/_layout.tsx b/app/_layout.tsx
index 97716e9..0986b51 100644
--- a/app/_layout.tsx
+++ b/app/_layout.tsx
@@ -11,17 +11,20 @@ SplashScreen.preventAutoHideAsync();
 
 export default function RootLayout() {
   const colorScheme = useColorScheme();
-  const [loaded] = useFonts({
+  const [loaded, error] = useFonts({
     SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
   });
 
   useEffect(() => {
-    if (loaded) {
+    if (loaded || error) {
       SplashScreen.hideAsync();
+      if (error) {
+        console.warn(`Error in loading fonts: ${error}`);
+      }
     }
-  }, [loaded]);
+  }, [loaded, error]);
 
-  if (!loaded) {
+  if (!loaded && !error) {
     return null;
   }
 
```

**Comment by douglowder** [comment]
Examples have been updated to fix the issue, see comments in #29607.
